src/openstudio_lib/ConstructionObjectVectorController.cpp

```diff
@@ -149,75 +151,61 @@ void ConstructionObjectVectorController::insert(const OSItemId& itemId, int posi
       }
     }
 
-    if (position < 0) {
-      position = construction.numLayers();
+    if (insertPosition < 0) {
+      insertPosition = construction.numLayers();
     }
 
-    if (deleteExisting) {
-      //  Checking if from library or from model isn't sufficient, if from model it also HAS to be already in the layer list
-      std::vector<unsigned> existingIndices = construction.getLayerIndices(*material);
-      if (!existingIndices.empty()) {
-        int existingPos = existingIndices[0];
-        LOG(Debug, "Erasing layer at position = " << existingPos);
-        construction.eraseLayer(existingPos);
-      } else {
-        LOG(Debug, "Layer wasn't present to begin with");
-      }
+    if (construction.insertLayer(insertPosition, *material) && erasePosition) {
+      construction.eraseLayer(*erasePosition);
     }
-
-    construction.insertLayer(position, *material);
   }
 }
 
 void ConstructionObjectVectorController::onReplaceItem(OSItem* currentItem, const OSItemId& replacementItemId) {
   if (m_modelObject) {
 
+    // currentItem position is set by OSDropZone::setItemIds
+    boost::optional<int> toPosition = currentItem->position();
+    boost::optional<int> fromPosition = replacementItemId.position();
```

Comment:
```suggestion
    boost::optional<int> toPosition_ = currentItem->position();
    boost::optional<int> fromPosition_ = replacementItemId.position();
```

Reply:
Is this a standard you want to follow?  Trailing `_` on optionals? Or is that to indicate it is a local variable?

Reply:
I used _optional before which i liked better. But apparently anything starting with a _ is reserved for standard library implementers. So optional_ is what I use now. 
I find it easier to understand what is an optional that way